### ISSUE TITLE
implement nativeWebTap for Android

### ISSUE THREAD

**Issue by 0x1mason** [appium/appium#4075]
Related to https://github.com/appium/appium/pull/4005.

Workaround is deriving x.y coordinates from WV element and then passing them to the native context.

**Comment by sonton** [comment]
It has been one year since this bug open. Is there any update on it yet ? 
Thanks

**Comment by lujian2036** [comment]
when can solve this problem ?there are so much many mobile webelement  use tap instead of click event to respone user action on android chrome browse

**Comment by fabytta** [comment]
This need to be resolved somehow, and switching between WB and native contexts to get the coordinates and then tap on the position seems to be too much time consuming (due to several switches in the same page)
Any other solution or workaround

**Comment by TheCodeSharman** [comment]
I'm also in the situation where the application under test uses a WebView and a need to simulate tap events instead of click events. Are there any plans to implement this? Would be really helpful for me too.

**Comment by jlipps** [comment]
Yes, this is in our prioritized backlog so there are plans to implement it. It will still take some time to get to, however.

**Comment by anjanabendre** [comment]
Will you please update us on the fix of this issue. 
We have hybrid app, developed using ionic cordova platform. Most of the pages are in webview and tap/touch and drag and drop is required for all elements. Now its not working.

So our automation for this application is completely on hold.

Appreciate quick fix on these issues.

Thanks,
Anjana Bendre

**Comment by JackGarbiec** [comment]
I would also ask if it's in the plans to be completed any time soon? It takes a lot of 'sticking a plaster' solutions to get touch actions to work in an android web app.

**Comment by shibupanda** [comment]
@jlipps 
It may not be a right thread but I just like to know whether is there any function available where we can pass web element from webview and expect to return Native X and Y co-ordinate as Native. So with this I can bypass nativeWebTap capability as I don't want to tap every element with co-ordinate.

I am using Java for the testcases.

**Comment by jlipps** [comment]
@shibupanda no, there is no such functionality, but you could accomplish this on the client side by using the same algorithm that nativeWebTap uses, basically:

1. find the rect of the element in the webpage
2. find the rect for the webview containing the element
3. convert the web position of the element to a native position by translating the scale and position of the webview to that of the entire native screen.

**Comment by mykola-mokhnach** [comment]
Fixed by https://github.com/appium/appium-uiautomator2-driver/pull/1044
